### PR TITLE
ARTEMIS-3873 AMQP Broker Conn Encrypted Attrs

### PR DESCRIPTION
**Pull request by ryan-highley**
Adds support for ENC(...) attribute values for user and password on
amqp-connection.

### CODE REVIEW

File: artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/FileConfigurationBrokerConnectionEncryptedTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+   public void testDefaults() {
+      // empty
+   }
+
+   @Test
+   public void testAMQPBrokerConfigEncryptedUserAndPassword() {
+
+      List<AMQPBrokerConnectConfiguration> brokerConnections = conf.getAMQPConnection();
+      Assert.assertNotNull("brokerConnections is not null", brokerConnections);
+      Assert.assertTrue("brokerConnections is not empty", !brokerConnections.isEmpty());
```
Comment (brusdev) [inline artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/FileConfigurationBrokerConnectionEncryptedTest.java:44]: The assertion message should describe what's wrong if the condition isn't met
```suggestion
      Assert.assertNotNull("brokerConnections is null", brokerConnections);
      Assert.assertFalse("brokerConnections is empty", brokerConnections.isEmpty());
```

File: artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/FileConfigurationBrokerConnectionEncryptedTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            // Both the encrypted and plain user and password use the same expected value
+            Assert.assertEquals("testuser", brokerConnection.getUser());
+            Assert.assertEquals("testpassword", brokerConnection.getPassword());
+
+         }
+      }
+
+      Assert.assertTrue("enc-test configuration present", encTest);
+      Assert.assertTrue("plain-test configuration present", plainTest);
+      Assert.assertTrue("empty-test configuration present", emptyTest);
```
Comment (brusdev) [inline artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/FileConfigurationBrokerConnectionEncryptedTest.java:73]: The assertion message should describe what's wrong if the condition isn't met
```suggestion
      Assert.assertTrue("enc-test configuration is not present", encTest);
      Assert.assertTrue("plain-test configuration is not present", plainTest);
      Assert.assertTrue("empty-test configuration is not present", emptyTest);
```

File: artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/FileConfigurationBrokerConnectionEncryptedTest.java (hunk 1 above)
Comment (ryan-highley) [reply artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/FileConfigurationBrokerConnectionEncryptedTest.java:44]: Assertion messages stating the opposite of the assertion logic always seemed backwards to me, but that's also one of those perennial "spaces vs. tabs" debates.

**Review by brusdev — APPROVED** [review]
LGTM +1

**Comment by brusdev** [comment]
@ryan-highley can you squash your 3 commits?

**Comment by ryan-highley** [comment]
Wow. Guess the old squash skills against remote repos needed some extensive rust removal! Should be good now. Thanks for the review.

**Comment by brusdev** [comment]
@ryan-highley thanks for your contribution!

**Pull request merged**